v2/network/session.go

```diff
@@ -437,7 +445,7 @@ func (session *Session) Connect(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-	if connOption.SSL {
+	if connOption.SSL || connOption.TLSConfig != nil {
```

Comment:
i believe `session.negotiate` is called several other places like`pckType == RESEND `, do we need to update those?

Reply:
It look like relaying on connOption.SSL  is enough. the `connOption.SSL` is set based on connection string protocol schema and in case of TLS config it always need to be set to tcps so we are sure that `connOption.SSL == true" when `connOption.TLSConfig != nil'